### PR TITLE
added initial draft for input config file

### PR DESCRIPTION
**Pull request by Nikhil-dot-Kumar**
@pramurthy @msrivathsa @upendrasahu , please review the input config file.

### CODE REVIEW

File: documentation/design/drafts/input.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+providers:
+  github:       # type of provider , can be butbucket or gitlabs also or normal private self hosted git 
+  - account_name: maplelabs     #github account username
+    periodicity: 300        #preiodicity for pulling changes using github APIs
+    credentials:
+      username:     #either username or email is needed (mandatory)
+      email:
+      token: XXX        #github access token to be used as it is current github auth standard 
+    repo_list:      #list of repository names
+    - github_audit
```
Comment (pramurthy) [inline documentation/design/drafts/input.md:16]: These repositories can be under  other user account. Better to use full URL

Comment (msrivathsa) [inline documentation/design/drafts/input.md:16]: These repositories can be under different organization, users. Better to take full URL

Comment (msrivathsa) [reply documentation/design/drafts/input.md:16]: Better to organize in the below hierarchy.

Git Audit Jobs:
     Each Job configuration.
             Job Specific configuration: Name, interval, meta data etc which are in the context of this program.
             repo type: GitHub
             repo specific configuration: path, url, credentials for login specific to target repo.
             Job output configuration: 

System level configuration
targets

Example:

auditJobs:
        - Name:    apache-project-progress
          polling_interval: 300
          tags:
                    tag1: one
          repo_type: GitHub
          repo_name:  apache2
          repo_config:
                access_token:   XXXXXXKXKX
          output:
             - target_name: es1

targets:
     #configuration related to es1 used above.

**Review by msrivathsa — CHANGES_REQUESTED** [review]
Specific suggestions mentioned above for reorganization of the config file section.

**Comment by Nikhil-dot-Kumar** [comment]
@msrivathsa @pramurthy @upendrasahu  , added output json file also. Please review.

File: documentation/design/drafts/input.md (hunk 1 above)
Comment (Nikhil-dot-Kumar) [reply documentation/design/drafts/input.md:16]: @msrivathsa , sure updating input yaml file as suggested

Comment (Nikhil-dot-Kumar) [reply documentation/design/drafts/input.md]: @msrivathsa , added the changes in input.md file as suggested , please review

File: documentation/design/drafts/output.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Github Audit Plugin Output Json Response
+
+Github audit plugin will be providing output in following json format:
+
+## Commits related
+### Type: commit 
+```json
+{
+    "type": "commit",
```
Comment (msrivathsa) [inline documentation/design/drafts/output.md:9]: Repo information fields are missing. For example, repo_name

**Review by msrivathsa — CHANGES_REQUESTED** [review]

File: documentation/design/drafts/output.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+```json
+{
+    "type": "pull_request",
+    "repo_type":"github",
+    "created_at": "2022-08-30T16:25:04Z",
+    "updated_at": "2022-08-30T16:25:04Z",
+    "closed_at": "",
+    "state": "open",
+    "pull_request_no": "1",
+    "title": "initial PR",
```
Comment (msrivathsa) [inline documentation/design/drafts/output.md:33]: Test

Comment (msrivathsa) [reply documentation/design/drafts/output.md:33]: It was a test comment. Review is resolved

File: documentation/design/drafts/output.md (hunk 2 above)
Comment (Nikhil-dot-Kumar) [reply documentation/design/drafts/output.md]: Added required information

**Comment by Nikhil-dot-Kumar** [comment]
@pramurthy @msrivathsa @upendrasahu  added design question file , Please review and add more questions .

File: documentation/design/output.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ Github audit plugin will be providing output in following json format:
 {
     "type": "commit",
```
Comment (pramurthy) [inline documentation/design/output.md:9]: May be we should follow snappyflow convention of calling it as documenType

File: documentation/design/output.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ Github audit plugin will be providing output in following json format:
...
     "repo_type":"github",
+    "repo_name":"test_repo",
+    "repo_url":"https://github.com/testurl",
     "created_at": "2022-08-30T16:25:04Z",
     "message": "initial PR",
     "url": "https://api.github.com/repos/pramurthy/sf-apm-agent/commits/9a5a338a2b6f9d435faa9adbda1f952276c1aea8",
```
Comment (pramurthy) [inline documentation/design/output.md:15]: Do we get user who made the commit? If so we should include it.

File: documentation/design/output.md [hunk 6]
```diff
@@ -8,6 +8,8 @@ Github audit plugin will be providing output in following json format:
 {
     "type": "commit",
     "repo_type":"github",
+    "repo_name":"test_repo",
+    "repo_url":"https://github.com/testurl",
     "created_at": "2022-08-30T16:25:04Z",
```
Comment (pramurthy) [inline documentation/design/output.md:13]: Better to include time as timestamp. It is upto the user to display in their required format.

File: documentation/design/output.md (hunk 4 above)
Comment (msrivathsa) [inline documentation/design/output.md:9]: May be we should follow snappyflow convention of calling it as documenType

File: documentation/design/output.md (hunk 6 above)
Comment (msrivathsa) [inline documentation/design/output.md:13]: Better to include time fields as epoch timestamp. It is upto the user to display in their required format.

File: documentation/design/output.md (hunk 5 above)
Comment (msrivathsa) [inline documentation/design/output.md:15]: Do we get user who made the commit? If so we should include it.

**Review by msrivathsa — APPROVED** [review]

**Pull request merged**